proposals/0000-splice-imports.rst

```diff
@@ -0,0 +1,319 @@
+.. author:: Matthew Pickering
+.. date-accepted:: Leave blank. This will be filled in when the proposal is accepted.
+.. ticket-url:: Leave blank. This will eventually be filled with the
+                ticket URL which will track the progress of the
+                implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/412>`_.
+.. contents::
+.. sectnum::
+
+Explicit Splice Imports Extension
+==============
+
+This proposes a new extension, ``ExplicitSpliceImports``, which modifies the
+import syntax so that imports which are only used at compile-time are marked explicitly.
+
+
+Motivation
+----------
+
+The primary goal of this proposal is to distinguish, in the module header, three different ways that
+imported module imports are used:
+
+1. Imported modules whose code is executed only at compile time;
+2. Imported modules whose code is executed only at runtime;
+3. Imported modules whose code is executed both at compile time and runtime.
+
+Distinguishing these 3 different cases has several advantages:
+
+1. Currently, if a module enables ``TemplateHaskell``, then all imported modules
+   are compiled to object code before name resolution takes place. The ensures that any top level splices that may be encountered are able to be fully evaluated.
+   This is a pessimisation because most of the imported identifiers, which we have taken such pains to ensure we can run, will not
+   actually be used in a top-level splice.
+   Proposals (such as `#14905 <https://gitlab.haskell.org/ghc/ghc/-/issues/14095>`_) to increase build parallelism are far less effective
+   in projects which use ``TemplateHaskell`` because name resolution depends on code generation
+   for all dependencies.
+   By distinguishing imported modules whose code is executed only at runtime
+   (which in common cases will be a small fraction of imported modules), we are
```

Comment:
Did you mean to say that a small fraction of imported modules will be used only at compile-time? (Currently the text says “at runtime”)